src/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { createContext, useContext, useState } from "react";
+
+type AuthContextType = {
+  isLoggedIn: boolean;
+  userEmail: string | null;
+  login: (email: string) => void;
+  logout: () => void;
+};
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+export const AuthProvider = ({ children }: { children: React.ReactNode }) => {
+  const [isLoggedIn, setIsLoggedIn] = useState<boolean>(() => {
+    return localStorage.getItem("isLoggedIn") === "true";
+  });
+  const [userEmail, setUserEmail] = useState<string | null>(() => {
+    return localStorage.getItem("userEmail");
+  });
+
+  const login = (email: string) => {
+    setIsLoggedIn(true);
+    setUserEmail(email);
+    localStorage.setItem("isLoggedIn", "true");
+    localStorage.setItem("userEmail", email);
+  };
+
+  const logout = () => {
+    setIsLoggedIn(false);
+    setUserEmail(null);
+    localStorage.removeItem("isLoggedIn");
+    localStorage.removeItem("userEmail");
+  };
+
+  return (
+    <AuthContext.Provider value={{ isLoggedIn, userEmail, login, logout }}>
+      {children}
+    </AuthContext.Provider>
+  );
+};
+
+export const useAuth = () => {
+  const context = useContext(AuthContext);
+  if (!context) throw new Error("useAuth must be used within an AuthProvider");
+  const { userEmail, ...rest } = context;
+  const userName = userEmail ? userEmail.split("@")[0] : null;
```

Comment:
문자열 조작처럼 가독성을 저해하는 코드는 유틸함수로 추상화해도 좋을 것 같아요.

```suggestion
  const userName = getUserNameFromEmail(userEmail);
```